refactor: update `blas/ext/base/gfill-by` to follow current project conventions

headlessNode opened the pull request:
## Description

> What is the purpose of this pull request?

This pull request:

-   update the package to follow current project conventions

## Related Issues

> Does this pull request have any related issues?

No. 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

headlessNode committed:
refactor: update to follow current project conventions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

headlessNode committed:
refactor: update to follow current project conventions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

headlessNode committed:
Merge branch 'stdlib-js:develop' into refactor-gfill-by

headlessNode committed:
Merge branch 'stdlib-js:develop' into refactor-gfill-by

headlessNode committed:
Merge branch 'refactor-gfill-by' of https://github.com/headlessNode/stdlib into refactor-gfill-by

headlessNode committed:
refactor: apply code review suggestion

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/gfill-by/README.md:
```diff
@@ -48,18 +48,11 @@ gfillBy( x.length, x, 1, fill );
 -   **clbk**: callback function.
 -   **thisArg**: execution context (_optional_).
 
-The invoked callback is provided four arguments:
```
@headlessNode Is there a reason you moved this list until after the `thisArg` example?

headlessNode replied:
@kgryte The `thisArg` is a parameter of `gfill-by`, and I felt it would be clearer if the parameter list and their corresponding descriptions were presented together. 

Currently, the parameters of `gfill-by` are listed first, followed by the parameters of the `callback`, and then the description of `thisArg`.

kgryte replied:
Yeah, I understand that tension. That said, IIRC, we are fairly (?) consistent across the code base in using the existing order. E.g., see https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/ndarray/base/unary-by and https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/strided/base/map-by.

headlessNode replied:
Hmmm... I'll revert the change then. Thanks!

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/gfill-by/README.md:
```diff
@@ -174,20 +170,12 @@ gfillBy.ndarray( 3, x, 1, x.length-3, fill );
 var Float64Array = require( '@stdlib/array/float64' );
 var gfillBy = require( '@stdlib/blas/ext/base/gfill-by' );
 
 function fill() {
```
Instead of this wrapper function, you could use `discreteUniform.factory( -100, 100 )`.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/gfill-by/benchmark/benchmark.js:
```diff
@@ -21,16 +21,26 @@
 
 var bench = require( '@stdlib/bench' );
-var uniform = require( '@stdlib/random/base/uniform' ).factory;
+var discreteUniform = require( '@stdlib/random/base/discrete-uniform' );
```
I'd stick with `uniform` here, as it is lighter weight than `discrete-uniform`. And instead of `rand` below, you can stick with using the `factory` method, as done previously. In general, this benchmark could be left as is and be fine.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/gfill-by/benchmark/benchmark.ndarray.js:
```diff
@@ -21,16 +21,26 @@
 
 var bench = require( '@stdlib/bench' );
-var uniform = require( '@stdlib/random/base/uniform' ).factory;
+var discreteUniform = require( '@stdlib/random/base/discrete-uniform' );
```
Same comments.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/gfill-by/examples/index.js:
```diff
@@ -18,20 +18,12 @@
-		sign = 1.0;
-	}
-	return sign * rand;
+	return discreteUniform( -100, 100 );
```
Same comments as README. No need for `fill` wrapper.

headlessNode committed:
refactor: apply code review suggestions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

headlessNode commented:
@kgryte Changes addressed.

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gfill-by/README.md:
````diff
@@ -174,26 +169,14 @@ gfillBy.ndarray( 3, x, 1, x.length-3, fill );
 ```javascript
-var round = require( '@stdlib/math/base/special/round' );
-var randu = require( '@stdlib/random/base/randu' );
+var discreteUniform = require( '@stdlib/random/base/discrete-uniform' );
````
```suggestion
var discreteUniform = require( '@stdlib/random/base/discrete-uniform' ).factory();
```

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gfill-by/README.md:
```diff
@@ -174,26 +169,14 @@ gfillBy.ndarray( 3, x, 1, x.length-3, fill );
 console.log( x );
 
-gfillBy( x.length, x, 1, fill );
+gfillBy( x.length, x, 1, discreteUniform.factory( -100, 100 ) );
```
```suggestion
gfillBy( x.length, x, 1, discreteUniform( -100, 100 ) );
```

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gfill-by/examples/index.js:
```diff
@@ -18,24 +18,12 @@
 
-var round = require( '@stdlib/math/base/special/round' );
-var randu = require( '@stdlib/random/base/randu' );
+var discreteUniform = require( '@stdlib/random/base/discrete-uniform' );
```
```suggestion
var discreteUniform = require( '@stdlib/random/base/discrete-uniform' ).factory();
```

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gfill-by/examples/index.js:
```diff
@@ -18,24 +18,12 @@
 console.log( x );
 
-gfillBy( x.length, x, 1, fill );
+gfillBy( x.length, x, 1, discreteUniform.factory( -100, 100 ) );
```
```suggestion
gfillBy( x.length, x, 1, discreteUniform( -100, 100 ) );
```

headlessNode committed:
docs: apply review suggestion

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode committed:
docs: apply review suggestion

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode committed:
docs: apply review suggestion

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode committed:
docs: apply review suggestion

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode committed:
fix: factory invocation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/gfill-by/benchmark/benchmark.js:
```diff
@@ -43,15 +43,12 @@ function createBenchmark( len ) {
 		b.tic();
 		for ( i = 0; i < b.iterations; i++ ) {
-			y = gfillBy( x.length, x, 1, f );
+			y = gfillBy( x.length, x, 1, uniform( -100, 100 ) );
```
This change is not desired, as now you are also measuring how quickly `uniform` can return a function.

kgryte committed:
Discard changes to lib/node_modules/@stdlib/blas/ext/base/gfill-by/benchmark/benchmark.js

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/gfill-by/benchmark/benchmark.ndarray.js:
```diff
@@ -43,15 +43,12 @@ function createBenchmark( len ) {
 		b.tic();
 		for ( i = 0; i < b.iterations; i++ ) {
-			y = gfillBy( x.length, x, 1, 0, f );
+			y = gfillBy( x.length, x, 1, 0, uniform( -100, 100 ) );
```
Same thing.

kgryte committed:
Discard changes to lib/node_modules/@stdlib/blas/ext/base/gfill-by/benchmark/benchmark.ndarray.js

kgryte reviewed (approved):
LGTM.
